### ISSUE TITLE
Improvements

### ISSUE THREAD

**Issue by TheCodeDestroyer** [ormjs/orm.js#4]
*IDEAS:*

* Right now there are too many libs that might not be needed each time package is installed so I would break it down into different repositories
* Improve documentation move it maybe to GitHub wiki and fix styling of code examples
* Add build and dependency status to README.md
* Republish bower and npm under new url
* ...

**Comment by darthtrevino** [comment]
I think the most pressing need is for a solid refactoring of the architecture while maintaining, as best we can, the existing API surface area. Test coverage is not great, and a lot of features are kind of buggy (like JSON dumps, the in-memory store didn't even work initially in tests.).

As far as package structures, there could be:
ormjs
ormjs-store-XXX (mysql, sqlite, etc..)
ormjs-sync
PassportJS is a good example of a project with multiple published packages that work together. 

persistence.js is a monster - 2k+ lines; browser, AMD, and NodeJS hooks. I think we could refactor it into a Facade that thunks over to a federation of small objects. For instance, the session data should be in a separate session-management class. I also think the hooks wiring it into the browser as a global or into module.exports should be separate from the core persistence code. It would break back-compat to a degree, but we could use something like browserify and write the core code to just target node.

Eventually I'd like to see promise support; either as a v2 refactor or a new interface. This is one of my gripes with the API and it's an issue that's commonly requested in persistence.js.

We should probably remove or update the flattr link in the Readme.

**Comment by TheCodeDestroyer** [comment]
I agree, like I already mentioned in the comment of your commit. Also I removed the flatter for now,
maybe we can add gittip or flatter later...
One more thing we should decide upon a coding style.
I would suggest [idiomatic.js](https://github.com/rwaldron/idiomatic.js) or the [Google Guides](https://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml), which one would you prefer?
